src/praisonai-agents/praisonaiagents/process/process.py

```
@@ -1065,35 +1065,40 @@ async def ahierarchical(self) -> AsyncGenerator[str, None]:
+                    tid for tid, tk in self.tasks.items() if tk.name != "manager_task"
+                ]
                 logging.warning(
                     f"Manager selected invalid task_id={selected_task_id} "
                     f"(attempt {invalid_selection_attempts}/{MAX_INVALID_SELECTIONS}); valid IDs: {valid_task_ids}"
                 )
                 # Set error context for next iteration (instead of appending to prompt that gets rebuilt)
                 error_context = (
                     f"\n\n[ERROR] Your previous selection of task_id={selected_task_id} was invalid. "
-                    f"Valid task IDs are: {valid_task_ids}. Please select again from the valid options."
+                    f"Valid task IDs are: {valid_task_ids}. Never select manager_task. "
+                    f"Please select again from the valid options."
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)